### PR TITLE
 Make GrantConditionOnAttack PausableConditional 

### PR DESCRIPTION
**Pull request by MustaphaTR**
Disabling it revokes the condition and makes it not possible to be granted while it is disabled.

Pausing just makes it impossible to grant. Existing conditions will be revoked by timer.

Testcase makes it so Light Tank gets gatling like weapon system changing between 1TNK, 2TNK and 3TNK weapons. Normally takes 8 shots to go up, but when elite it takes 4.

### CODE REVIEW

**Comment by pchote** [comment]
What is the benefit of this vs writing an expression that combines the output of `GrantConditionOnAttack` with the condition that would disable/pause it?

**Comment by MustaphaTR** [comment]
Why i need is for is pretty much the testcase. When elite Gatling Tank moves thru the levels faster.

**Comment by GraionDilach** [comment]
Different name between promoted and unpromoted weapon with GCOA tracking per-armament names/additional dummy weapons enabled with veterancy... I can think of ways to set this up already.

**Comment by MustaphaTR** [comment]
I guess Graion is right, tho i don't see any reason for not making the trait conditional.

**Review by Mailaender — DISMISSED** [review]
Code looks fine.

File: OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnAttack.cs [hunk 1]
```diff
@@ -59,11 +58,9 @@ public class GrantConditionOnAttack : INotifyCreated, ITick, INotifyAttack
 		Target lastTarget = Target.Invalid;
 
 		public GrantConditionOnAttack(ActorInitializer init, GrantConditionOnAttackInfo info)
-		{
-			this.info = info;
-		}
+			: base(info) { }
 
-		void INotifyCreated.Created(Actor self)
+		protected override void Created(Actor self)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnAttack.cs]: This is missing a call to `base.Created(self);`.

**Review by pchote — COMMENTED** [review]
Noticed one obvious code issue, and can you please rebase this on current bleed to allow proper testing?

**Comment by MustaphaTR** [comment]
Updated and rebased.

**Review by abcdefg30 — DISMISSED** [review]

**Comment by abcdefg30** [comment]
Testcase needs to be removed.

**Comment by MustaphaTR** [comment]
Removed testcase.

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/8bf14ef1d31f7a7952aa7819ba4b4df4b4397bf2...f423d19f41d9c6220f6f7cc95658f4b094323078)

**Pull request merged**